Drivers: NXP :  dependency for CONFIG_NXP_LP_FLEXCOMM missing hal include driver_lpflexcomm

**Describe the bug**

in https://github.com/zephyrproject-rtos/zephyr/pull/85446 
there is one issue revealed for driver dependency

if we do not define the flexcomm_i2c/flexcomm_uart/flexcomm_i2s, the fsl_lpflexcomm.h will not included, but the /mfd/nxp_lp_flexcomm.h still use it

for mimxrt700_evk/mimxrt798s/cm33_cpu0
```
/home/shared/disk/zephyr_project/zephyr_test/zephyr/include/zephyr/drivers/mfd/nxp_lp_flexcomm.h:9:10: fatal error: fsl_lpflexcomm.h: No such file or directory
    9 | #include "fsl_lpflexcomm.h"
      |  
```

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
     - mimxrt700_evk/mimxrt798s/cm33_cpu0
     - frdm_mcxn947/mcxn947/cpu1
     - frdm_mcxn947/mcxn947/cpu0
     - frdm_mcxn236/mcxn236
 - What have you tried to diagnose or workaround this issue?
 - build analysis
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 - No

**To Reproduce**

```
west twister -p mimxrt700_evk/mimxrt798s/cm33_cpu0 -T samples/basic/minimal/
```

**Expected behavior**

build pass

**Impact**

minimal checking and driver dependency

**Logs and console output**

```
In file included from /home/shared/disk/zephyr_project/zephyr_test/zephyr/drivers/mfd/mfd_nxp_lp_flexcomm.c:14:
/home/shared/disk/zephyr_project/zephyr_test/zephyr/include/zephyr/drivers/mfd/nxp_lp_flexcomm.h:9:10: fatal error: fsl_lpflexcomm.h: No such file or directory
    9 | #include "fsl_lpflexcomm.h"
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.

```

**Environment (please complete the following information):**

 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: b10587f2d6190dbc8897702540bce63b9d448cc4

--- a/west.yml
+++ b/west.yml
@@ -203,7 +203,7 @@ manifest:
       groups:
         - hal
     - name: hal_nxp
-      revision: 3fc5f811fb8c0bdd90a8d965c50707172ce80d05
+      revision: 9dc7449014a7380355612453b31be479cb3a6833
       path: modules/hal/nxp
       groups:
         - hal

